api/awsauth/awsauth/auth_app.py

```diff
@@ -200,16 +239,21 @@ def register_edge(event, context):
     headers = {
         "access-control-allow-origin": [{"key": "Access-Control-Allow-Origin", "value": "*"}],
     }
-    if "email" not in data:
-        return {"status": 400, "errorMessage": "Missing email parameter", "headers": headers}
 
-    email = data["email"]
-    if not re.match(EMAIL_REGEX, email):
-        return {"status": 400, "errorMessage": "Invalid email", "headers": headers}
+    try:
+        args = RegistrationArguments.make_from_json_body(data)
+    except ValueError as exn:
+        return {"status": 400, "errorMessage": str(exn), "headers": headers}
```

Comment:
dumping str(exn) into a message that may be returned to the client seems way more dangerous than a static string. I don't know much about the context here or security but enough to be paranoid.  I'd make an exception type per message or have some attribute of an exception that is known to be safe.